WordPress/Classes/ViewRelated/NUX/LoginViewController.m

```diff
@@ -381,6 +387,51 @@ - (void)forgotPassword:(id)sender
     [[UIApplication sharedApplication] openURL:forgotPasswordURL];
 }
 
+- (IBAction)findLoginFromOnePassword:(id)sender
+{
+    if (_userIsDotCom == false && _siteUrlText.text.isEmpty) {
+        [self displayOnePasswordEmptySiteAlert];
+        return;
+    }
+ 
+    NSString *loginURL = _userIsDotCom ? WPOnePasswordWordPressComURL : _siteUrlText.text;
+    
+    [[OnePasswordExtension sharedExtension] findLoginForURLString:loginURL
+                                                forViewController:self
+                                                           sender:sender
+                                                       completion:^(NSDictionary *loginDict, NSError *error) {
+        if (!loginDict) {
+            if (error.code != AppExtensionErrorCodeCancelledByUser) {
+                DDLogError(@"OnePassword Error: %@", error);
+            }
+            return;
+        }
+
+        _usernameText.text = loginDict[AppExtensionUsernameKey];
+        _passwordText.text = loginDict[AppExtensionPasswordKey];
+        [self signIn];
+    }];
+}
+
+
+#pragma mark - One Password Helpers
+
+- (void)displayOnePasswordEmptySiteAlert
+{
+    NSString *message = NSLocalizedString(@"Site address is required to be entered before 1Password can be used.",
```

Comment:
Maybe the message wording could be a little better? Maybe just say "A site address is required before..."?  The "to be entered" sounds a bit klunky to me.  :shrug: